### ISSUE TITLE
Error occurred while FactoryBot/AssociationStyle cop was inspecting the file

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#11883]
I have the following code:

```ruby
FactoryBot.define do
  factory :user do
    password { Faker::Internet.password }
    email { Faker::Internet.email }
  end

  factory :admin, parent: :user, traits: %i[admin]
end
```

## Actual behavior

```
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop -d spec/factories/users.rb  --only FactoryBot/AssociationStyle
For /app: configuration from /app/.rubocop.yml
configuration from /usr/local/bundle/gems/rubocop-capybara-2.18.0/config/default.yml
Default configuration from /usr/local/bundle/gems/rubocop-1.50.2/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-capybara-2.18.0/lib/../config/default.yml
configuration from /usr/local/bundle/gems/rubocop-factory_bot-2.23.0/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-factory_bot-2.23.0/lib/../config/default.yml
configuration from /usr/local/bundle/gems/rubocop-performance-1.17.1/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-performance-1.17.1/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-rails-2.19.1/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-rails-2.19.1/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-rake-0.6.0/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-rake-0.6.0/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-rspec-2.22.0/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-rspec-2.22.0/config/default.yml
Inheriting configuration from /app/.rubocop_todo.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /app/spec/factories/users.rb
An error occurred while FactoryBot/AssociationStyle cop was inspecting /app/spec/factories/users.rb:9:2.
undefined method `body' for #<RuboCop::AST::Node:0x000055bb01676468>
/usr/local/bundle/gems/rubocop-factory_bot-2.23.0/lib/rubocop/cop/factory_bot/association_style.rb:149:in `children_of_factory_block'
/usr/local/bundle/gems/rubocop-factory_bot-2.23.0/lib/rubocop/cop/factory_bot/association_style.rb:141:in `bad_associations_in'
/usr/local/bundle/gems/rubocop-factory_bot-2.23.0/lib/rubocop/cop/factory_bot/association_style.rb:58:in `on_send'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/cop/commissioner.rb:143:in `public_send'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/cop/commissioner.rb:143:in `block (2 levels) in trigger_restricted_cops'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/cop/commissioner.rb:171:in `with_cop_error_handling'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/cop/commissioner.rb:142:in `block in trigger_restricted_cops'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/cop/commissioner.rb:141:in `each'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/cop/commissioner.rb:141:in `trigger_restricted_cops'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/cop/commissioner.rb:70:in `on_send'
/usr/local/bundle/gems/rubocop-ast-1.28.1/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/usr/local/bundle/gems/rubocop-ast-1.28.1/lib/rubocop/ast/traversal.rb:137:in `each'
/usr/local/bundle/gems/rubocop-ast-1.28.1/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/usr/local/bundle/gems/rubocop-ast-1.28.1/lib/rubocop/ast/traversal.rb:158:in `on_block'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/cop/commissioner.rb:71:in `on_block'
/usr/local/bundle/gems/rubocop-ast-1.28.1/lib/rubocop/ast/traversal.rb:20:in `walk'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/cop/commissioner.rb:87:in `investigate'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/cop/team.rb:156:in `investigate_partial'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/cop/team.rb:98:in `investigate'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/runner.rb:345:in `block in inspect_file'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/runner.rb:344:in `each'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/runner.rb:344:in `flat_map'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/runner.rb:344:in `inspect_file'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/runner.rb:287:in `block in do_inspection_loop'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/runner.rb:321:in `block in iterate_until_no_changes'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/runner.rb:314:in `loop'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/runner.rb:314:in `iterate_until_no_changes'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/runner.rb:283:in `do_inspection_loop'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/runner.rb:164:in `block in file_offenses'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/runner.rb:189:in `file_offense_cache'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/runner.rb:163:in `file_offenses'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/runner.rb:154:in `process_file'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/runner.rb:135:in `block in each_inspected_file'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/runner.rb:134:in `each'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/runner.rb:134:in `reduce'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/runner.rb:134:in `each_inspected_file'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/runner.rb:120:in `inspect_files'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/runner.rb:73:in `run'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/cli/command.rb:11:in `run'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/cli/environment.rb:18:in `run'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/cli.rb:118:in `run_command'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/cli.rb:125:in `execute_runners'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/cli.rb:51:in `block in run'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/cli.rb:77:in `profile_if_needed'
/usr/local/bundle/gems/rubocop-1.50.2/lib/rubocop/cli.rb:43:in `run'
/usr/local/bundle/gems/rubocop-1.50.2/exe/rubocop:19:in `block in <top (required)>'
/usr/local/lib/ruby/2.7.0/benchmark.rb:308:in `realtime'
/usr/local/bundle/gems/rubocop-1.50.2/exe/rubocop:19:in `<top (required)>'
/usr/local/bundle/bin/rubocop:23:in `load'
/usr/local/bundle/bin/rubocop:23:in `<main>'
.

1 file inspected, no offenses detected

1 error occurred:
An error occurred while FactoryBot/AssociationStyle cop was inspecting /app/spec/factories/users.rb:9:2.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.50.2 (using Parser 3.2.1.0, rubocop-ast 1.28.1, running on ruby 2.7.8) [x86_64-linux]
Finished in 0.546730096997635 seconds
```

## RuboCop version

```
ydakuka@yauhenid:~/Work/main_app$ bin/rails_docker rubocop -V
1.50.2 (using Parser 3.2.1.0, rubocop-ast 1.28.1, running on ruby 2.7.8) [x86_64-linux]
  - rubocop-capybara 2.18.0
  - rubocop-performance 1.17.1
  - rubocop-rails 2.19.1
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.22.0
```
